migrations/default/3_test_views/up.sql

```diff
@@ -0,0 +1,63 @@
+create or replace view people_cris_view as (
+    select
+        db.crashes_cris.crash_id as crash_id,
+        db.units_cris.id as unit_id,
+        db.units_cris.unit_nbr,
+        db.people_cris.id as person_id,
+        db.people_cris.prsn_nbr,
+        db.people_cris.is_primary as is_primary,
+        db.unit_types.description as unit_type_description,
+        road_types.description as road_type_description
+    from
+        db.people_cris
+    left join db.units_cris on db.units_cris.id = db.people_cris.unit_id
+    left join
+        db.crashes_cris
+        on db.units_cris.crash_id = db.crashes_cris.crash_id
+    left join db.road_types on db.crashes_cris.road_type_id = road_types.id
+    left join db.unit_types on db.units_cris.unit_type_id = db.unit_types.id
+    order by crash_id asc, unit_id asc, person_id asc
+);
+
+create or replace view people_vz_view as (
+    select
+        db.crashes_vz.crash_id as crash_id,
+        db.crashes_vz.road_type_id,
+        db.units_vz.id as unit_id,
+        db.units_vz.unit_nbr,
+        db.units_vz.unit_type_id,
+        db.people_vz.id as person_id,
+        db.people_vz.prsn_nbr,
+        db.people_vz.is_primary as is_primary,
+        db.unit_types.description as unit_type_description,
+        road_types.description as road_type_description
+    from
+        db.people_vz
+    left join db.people_cris on db.people_vz.id = db.people_cris.id
+    left join db.units_vz on db.units_vz.id = db.people_cris.unit_id
+    left join db.units_cris on db.units_vz.id = db.units_cris.id
+    left join db.crashes_cris on db.units_cris.crash_id = db.crashes_cris.crash_id
+    left join db.crashes_vz on db.crashes_cris.crash_id = db.crashes_vz.crash_id
+    left join db.road_types on db.crashes_vz.road_type_id = road_types.id
+    left join db.unit_types on db.units_vz.unit_type_id = db.unit_types.id
+    order by crash_id asc, unit_id asc, person_id asc
+);
```

Comment:
it turns out that querying a transparent view of what VZ edits are applied across an entire chain of records (person > unit > crash) is pretty verbose. this is because our `_vz` tables (as currently designed) do not hold the keys we need to join records to together.

On the `people_vz` table, the `unit_id` is `null` by default. On the `units_vz` table, the `crash_id` column is `null` by default. 

This goes back to my dizziness around foreign keys in the `_vz` tables. I think this is the ideal setup, and we shouldn't need to write queries like this very often. but...yeah 🤷

Reply:
> Can we enforce that the FK relationships of child tables (people to units, units to crashes) are not null and require the target record to exist?

I believe I already have that covered for all the tables. The exception is the `_vz` tables, where we only enforce that the record's `id` column is a not-null foreign key taht references the table's corresponding `_cris` column.